bootstrap/src/main/java/jenkins/bootstrap/DependenciesTxt.java

```diff
@@ -0,0 +1,81 @@
+package jenkins.bootstrap;
+
+import hudson.util.VersionNumber;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Parses {@code dependencies.txt} produced by Maven dependency:list-dependencies mojo.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+class DependenciesTxt {
+    final List<Dependency> dependencies;
+    /**
+     * 'groupId:artifactId' to version number
+     */
+    private final Map<String,VersionNumber> jars = new HashMap<>();
+
+    public DependenciesTxt(InputStream i) throws IOException {
+        List<Dependency> list = new ArrayList<>();
+
+        try (Reader r = new InputStreamReader(i,"UTF-8");
+             BufferedReader br = new BufferedReader(r)) {
+
+            // line that we care about has 5 components: groupId:artifactId:packaging[:classifier]:version:scope
+            String line;
+            while ((line=br.readLine())!=null) {
```

Comment:
There must be misunderstanding. This is just to read `dependencies.txt` from a jar file.